app/web/features/auth/jail/PasswordSection.tsx

```diff
@@ -0,0 +1,116 @@
+import { TextField, Typography, useMediaQuery } from "@material-ui/core";
+import { Alert } from "@material-ui/lab";
+import Button from "components/Button";
+import { Empty } from "google-protobuf/google/protobuf/empty_pb";
+import { Error as GrpcError } from "grpc-web";
+import { useTranslation } from "next-i18next";
+import { useForm } from "react-hook-form";
+import { useMutation } from "react-query";
+import { service } from "service";
+import { theme } from "theme";
+import useChangeDetailsFormStyles from "features/auth/useChangeDetailsFormStyles";
+
+interface SetPasswordVariables {
+  password?: string;
+}
+
+interface SetPasswordFormData extends SetPasswordVariables {
+  passwordConfirmation?: string;
+}
+
+interface PasswordSectionProps {
+  updateJailed: () => void;
+  className?: string;
+}
+
+export default function PasswordSection({
+  updateJailed,
+  className,
+}: PasswordSectionProps) {
+  const classes = useChangeDetailsFormStyles();
+  const { t } = useTranslation(["auth", "global"]);
+
+  const isMdOrWider = useMediaQuery(theme.breakpoints.up("md"));
+
+  const {
+    errors,
+    getValues,
+    handleSubmit,
+    reset: resetForm,
+    register,
+  } = useForm<SetPasswordFormData>({
+    mode: "onBlur",
+  });
+  const onSubmit = handleSubmit(({ password }) => {
+    changePassword({ password });
+  });
+
+  const {
+    error: changePasswordError,
+    isLoading: isChangePasswordLoading,
+    isSuccess: isChangePasswordSuccess,
+    mutate: changePassword,
+    variables: setPasswordVariables,
+  } = useMutation<Empty, GrpcError, SetPasswordVariables>(
+    ({ password }) => service.jail.setPassword(password),
```

Comment:
Seeing type errors on this in the build - surprised the dev build doesn't complain about this.

```suggestion
  } = useMutation<void, GrpcError, SetPasswordVariables>(
    async ({ password }) => { await service.jail.setPassword(password); },
```
This should fix the type error